components/availability/src/Availability.svelte

```diff
@@ -334,27 +347,58 @@
     return epochs;
   }
 
-  $: days = scaleTime()
-    .domain([startDay, endDay])
-    .ticks(timeDay)
-    .filter((timestamp: Date) => {
-      if (show_weekends) {
-        return true;
-      } else {
-        return (
-          timestamp.toString() !== timeSaturday(timestamp).toString() &&
-          timestamp.toString() !== timeSunday(timestamp).toString()
-        );
+  function generateDayRange({
```

Comment:
My thinking in building it was that we may end up wanting this function to consider more props to customize it down the line; in practice, only the `reverse` prop ended up getting added toward the end of this.

I'll switch it to using ordered params instead of an object